File: Difficulty: Easy/Minimum Cost of ropes/minimum-cost-of-ropes.cpp
class Solution {
  public:
    int minCost(vector<int>& arr) {
        // code here
        priority_queue<int, vector<int>, greater<int>> pq;
        for(auto e: arr) pq.push(e);
        int ans = 0;
        while(pq.size()>1) {
            int a = pq.top();
            pq.pop();
            int b = pq.top();
            pq.pop();
            int sum = a+b;
            ans+=sum;
            pq.push(sum);
        }
        return ans;
    }
};

File: Difficulty: Medium/Minimum Cost of ropes/minimum-cost-of-ropes.java
class Solution {
    public static int minCost(int[] arr) {
        // code here
        int n = arr.length;
        Queue<Integer> pq = new PriorityQueue<>();
        for(int i: arr) pq.add(i);
        int ans = 0;
        while(pq.size()!=1) {
            int a = pq.poll();
            int b = pq.poll();
            ans+=(a+b);
            pq.add(a+b);
        }
        
        return ans;
    }
}